Tests/CSharp/TypeCastTests.cs

```diff
@@ -1313,11 +1313,11 @@ internal partial class CastTest
     }
     private decimal? Test2(object input)
     {
-        return (decimal?)input;
+        return (decimal?)(double?)input;
     }
     private decimal? Test2(int input)
     {
-        return input;
+        return (decimal?)(double?)input;
```

Comment:
Uglier but the source code wasn't better.
You can always run cleanups (which I guess most users do anyway after conversion) to remove code redundancies.